### PR TITLE
assembly: split mod into smaller files

### PR DESCRIPTION
**Pull request by hackaugusto**
## Describe your changes

as the title says.

this changes no code, just moves it around

## Checklist before requesting a review
- Repo forked and branch created from `next` according to naming convention.
- Commit messages and codestyle follow [conventions](./CONTRIBUTING.md).
- Relevant issues are linked in the PR description.
- Tests added for new functionality.
- Documentation/comments updated according to changes.

### CODE REVIEW

File: assembly/src/ast/module.rs [hunk 1]
```diff
@@ -0,0 +1,314 @@
...
+/// An abstract syntax tree of a Miden module.
+///
+/// A module AST consists of a list of procedure ASTs, a list of re-exported procedures, a list of
+/// imports, and module documentation. Local procedures could be internal or exported.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct ModuleAst {
+    pub(crate) local_procs: Vec<ProcedureAst>,
+    pub(crate) reexported_procs: Vec<ProcReExport>,
+    pub(crate) import_info: ModuleImports,
+    pub(crate) docs: Option<String>,
```
[the remark this answers is not in the document]
Comment (bobbinth) [reply assembly/src/ast/module.rs:26]: Would `pub(super)` be sufficient here?

File: assembly/src/ast/program.rs [hunk 2]
```diff
@@ -0,0 +1,329 @@
...
+/// A program AST consists of a body of the program, a list of internal procedure ASTs, a list of
+/// imported libraries, a map from procedure ids to procedure names for imported procedures used in
+/// the module, and the source location of the program.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct ProgramAst {
+    pub(crate) body: CodeBody,
+    pub(crate) local_procs: Vec<ProcedureAst>,
+    pub(crate) import_info: ModuleImports,
+    pub(crate) start: SourceLocation,
+}
```
Comment (bobbinth) [inline assembly/src/ast/program.rs:34]: Would `pub(super)` be sufficient here?

**Review by bobbinth — APPROVED** [review]
Looks good! Thank you! I left a couple of small comments inline.

File: assembly/src/ast/program.rs (hunk 2 above)
Comment (hackaugusto) [reply assembly/src/ast/program.rs]: changed

**Pull request merged**